samples/apps/copilot-chat-app/webapi/Skills/ChatSkill.cs

```diff
@@ -212,27 +215,116 @@ public async Task<string> AcquireExternalInformationAsync(SKContext context)
         if (plan.Steps.Count > 0)
         {
             SKContext planContext = await plan.InvokeAsync(plannerContext);
+            int tokenLimit = int.Parse(context["tokenLimit"], new NumberFormatInfo());
 
             // The result of the plan may be from an OpenAPI skill. Attempt to extract JSON from the response.
             if (!this.TryExtractJsonFromOpenApiPlanResult(planContext.Variables.Input, out string planResult))
             {
                 // If not, use result of the plan execution result directly.
                 planResult = planContext.Variables.Input;
             }
+            else
+            {
+                int relatedInformationTokenLimit = (int)Math.Floor(tokenLimit * this._promptSettings.RelatedInformationContextWeight);
+                planResult = this.OptimizeOpenApiSkillJson(planResult, relatedInformationTokenLimit, plan);
+            }
 
             string informationText = $"[START RELATED INFORMATION]\n{planResult.Trim()}\n[END RELATED INFORMATION]\n";
 
             // Adjust the token limit using the number of tokens in the information text.
-            int tokenLimit = int.Parse(context["tokenLimit"], new NumberFormatInfo());
             tokenLimit -= Utilities.TokenCount(informationText);
             context.Variables.Set("tokenLimit", tokenLimit.ToString(new NumberFormatInfo()));
 
             return informationText;
         }
-        else
+
+        return string.Empty;
+    }
+
+    /// <summary>
+    /// Try to optimize json from the planner response
+    /// based on token limit
+    /// </summary>
+    private string OptimizeOpenApiSkillJson(string jsonContent, int tokenLimit, Plan plan)
+    {
+        int jsonTokenLimit = (int)(tokenLimit * this._promptSettings.RelatedInformationContextWeight);
+
+        // Remove all new line characters + leading and trailing white space
+        jsonContent = Regex.Replace(jsonContent.Trim(), @"[\n\r]", string.Empty);
+        var document = JsonDocument.Parse(jsonContent);
+        string lastSkillInvoked = plan.Steps[^1].SkillName;
+        string lastSkillOperationInvoked = plan.Steps[^1].Name;
+
+        // Check if the last skill invoked was JiraSkill and deserialize the JSON content accordingly
+        if (string.Equals(lastSkillInvoked, "JiraSkill", StringComparison.Ordinal))
         {
-            return string.Empty;
+            Type skillResponseType = typeof(IssueResponse); // reasonable assumption of type so value isnt null
```

Comment:
Let's consider out the optimization to its own method (e.g., "OptimizeJiraSkillJson").